src/mlpack/methods/ann/layer/residual_impl.hpp

```diff
@@ -0,0 +1,209 @@
+/**
+ * @file methods/ann/layer/residual_impl.hpp
+ * @author Shubham Agrawal
+ *
+ * Implementation of the base class for neural network layers that are wrappers
+ * around other layers.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_LAYER_RESIDUAL_IMPL_HPP
+#define MLPACK_METHODS_ANN_LAYER_RESIDUAL_IMPL_HPP
+
+#include "residual.hpp"
+
+namespace mlpack {
+namespace ann {
+
+template<typename MatType>
+ResidualType<MatType>::ResidualType() :
+    MultiLayer<MatType>()
+{
+  // Nothing to do.
+}
+
+template<typename MatType>
+ResidualType<MatType>::ResidualType(const ResidualType& other) :
+    MultiLayer<MatType>(other)
+{
+  // Nothing to do here.
+}
+
+template<typename MatType>
+ResidualType<MatType>::ResidualType(ResidualType&& other) :
+    MultiLayer<MatType>(other)
+{
+  // Nothing to do here.
+}
+
+template<typename MatType>
+ResidualType<MatType>& ResidualType<MatType>::operator=(const ResidualType& other)
+{
+  if (this != &other)
+  {
+    MultiLayer<MatType>::operator=(other);
+  }
+
+  return *this;
+}
+
+template<typename MatType>
+ResidualType<MatType>& ResidualType<MatType>::operator=(ResidualType&& other)
+{
+  if (this != &other)
+  {
+    MultiLayer<MatType>::operator=(other);
+  }
+
+  return *this;
+}
+
+template<typename MatType>
+void ResidualType<MatType>::Forward(
```

Comment:
It would be a good idea here to add a comment about the general strategy here: we are passing the input through each of the layers separately, and then reducing the output.